### ISSUE TITLE
dFBT: Eth worker structure refactoring 

### ISSUE THREAD

**Issue by AnnaShaleva** [bane-labs/go-ethereum#29]
#### Rationale

After #28, dBFT activity is tightly bound to `OnTimeout()` events. dBFT  itself is a driver of consensus process, it relies on external timer routine and this scheme is poorly compatible with the existing Eth worker scheme. Eth worker itself is the entity that gathers mempooled transactions and submits new sealing tasks to dBFT via `Prepare()`, `FinalizeAndAssemble()` and `Seal()` callbacks. This scheme spawns unwanted synchronization mechanisms in `consensus/dbft/dbft.go` like `sealingLock` and `lastProposalLock`.

#### Suggestions

Treat this issue as kind of "epic".

Here's the desired way of `worker` functioning:
- [x] ~dBFT's timer routine should be moved outside the `consensus.Engine` implementation, ideally to the `miner/worker.go` level~ won't be done since current dBFT timer routine is independent from the miner/worker routines.
- [x] `worker` should submit new proposal according to the dBFT timer and only when dBFT needs new proposal. A dedicated issue for that is #54.

After proposed `worker` refactoring the following things can be improved:
- [x] ~Unify `chain` usages of the `DBFT` consensus engine (currently dBFT callbacks use `chain` provided via `Initialize` and Engine methods use the `chain` provided as an argument to the method).~ Nothing to do with it. If possible, external `chain` is used provided via `Engine` methods arguments (including block verification, thus, every node can verify block). If not possible, then `chain` from `DBFT` instance is used (for dBFT callbacks, for sealing task awaiting and for payloads verification), which is the same `chain` as the one provided via callbacks. So we're OK with the current situation.
- [x] ~Provide a way for DBFT to subscribe for new blocks *from P2P* and call `postBlock` callback not only when new block arrives from consensus, but also when new new block arrives via P2P (may be a part of #13)~ Implemented in #49.
- [x] Remove all notions of `epoch` and voting mechanism from `consensus/dbft/dbft.go` (accounts votes recalculations and etc.). A separate issue for this is #51.
- [x] ~Move `MixDigest` and `Nonce` block proposal's fields initialisation to the `Engine.Prepare` stage so that `Engine.Seal` can't change "the honest" block hash. After this we can completely replace `WorkerSealHash` implementation with `HonestSealHash`.~ Won't be done since dBFT routine should be independant from miner, and `Engine.Prepare` is called every time when miner needs to propose fresh task. dBFT can't provide relevant information in `Prepare` since it can always change afterwards in case of CV.
- [x] ~We likely don't need header's `Difficulty` calculations and this field is probably not relevant with dBFT.~ We need it, track in #58.

**Comment by AnnaShaleva** [comment]
Ref. https://github.com/bane-labs/go-ethereum/pull/28#discussion_r1396127241.

**Comment by AnnaShaleva** [comment]
Everything is done here, if some further dBFT-worker bunch enhancement/refactor is needed in future, we'll create separate issues for that. Currently we're OK with the existing implementation.
